packages/core/src/lighting/AmbientLight.ts

```diff
@@ -1,56 +1,139 @@
+  );
+  private static _mipLevelProperty: ShaderProperty = Shader.getPropertyByName("u_envMapLight.mipMapLevel");
+
+  private _scene: Scene;
+  private _diffuseSolidColor: Color = new Color(0.212, 0.227, 0.259);
+  private _diffuseIntensity: number = 1.0;
+  private _specularReflection: TextureCubeMap;
+  private _specularIntensity: number = 1.0;
+  private _diffuseMode: DiffuseMode = DiffuseMode.SolidColor;
+
+  /**
+   * Diffuse mode of ambient light.
+   */
+  get diffuseMode(): DiffuseMode {
+    return this._diffuseMode;
+  }
+
+  set diffuseMode(value: DiffuseMode) {
+    this._diffuseMode = value;
+    if (value === DiffuseMode.Texture) {
+      this._scene.shaderData.enableMacro(AmbientLight._diffuseMacro);
+    } else {
+      this._scene.shaderData.disableMacro(AmbientLight._diffuseMacro);
+    }
+  }
 
   /**
-   * Ambient light color.
+   * Diffuse reflection solid color.
+   * @remarks Effective when diffuse reflection mode is `DiffuseMode.SolidColor`.
    */
-  get color(): Color {
-    return this._color;
+  get diffuseSolidColor(): Color {
+    return this._diffuseSolidColor;
   }
 
-  set color(value: Color) {
-    this._color = value;
-    this.scene.shaderData.setColor(AmbientLight._colorProperty, this.lightColor);
+  set diffuseSolidColor(value: Color) {
+    if (value !== this._diffuseSolidColor) {
+      value.cloneTo(this._diffuseSolidColor);
+      this._scene.shaderData.setColor(AmbientLight._diffuseColorProperty, value);
+    }
   }
 
   /**
-   * Ambient light intensity.
+   * Diffuse reflection intensity.
    */
-  get intensity(): number {
-    return this._intensity;
+  get diffuseIntensity(): number {
+    return this._diffuseIntensity;
   }
 
-  set intensity(value: number) {
-    this._intensity = value;
-    this.scene.shaderData.setColor(AmbientLight._colorProperty, this.lightColor);
+  set diffuseIntensity(value: number) {
+    this._diffuseIntensity = value;
+    this._scene.shaderData.setFloat(AmbientLight._diffuseIntensityProperty, value);
   }
 
   /**
-   * Get the final light color.
+   * Specular reflection texture.
+   */
+  get specularTexture(): TextureCubeMap {
+    return this._specularReflection;
+  }
+
+  set specularTexture(value: TextureCubeMap) {
+    this._specularReflection = value;
+
+    const shaderData = this._scene.shaderData;
+
+    if (value) {
+      shaderData.setTexture(AmbientLight._specularTextureProperty, value);
+      shaderData.setFloat(AmbientLight._mipLevelProperty, this._specularReflection.mipmapCount);
+      shaderData.enableMacro(AmbientLight._specularMacro);
+    } else {
+      shaderData.disableMacro(AmbientLight._specularMacro);
+    }
+  }
+
+  /**
+   * Specular reflection intensity.
+   */
+  get specularIntensity(): number {
+    return this._specularIntensity;
+  }
+
+  set specularIntensity(value: number) {
+    this._specularIntensity = value;
+    this._scene.shaderData.setFloat(AmbientLight._specularIntensityProperty, value);
+  }
```

Comment:
Do we still need `specularColor` in ambientLight and shader ?

Reply:
`specularColor` can come form Materil, for example BlinnPhongMaterial. 
In `AmbientLight` must use `specularTexture`.

Reply:
done